### PR TITLE
Improve `RedundantInheritedCheck`

### PR DESCRIPTION
**Pull request by jgardn3r**
The redundant inherited check has a few issues:
* Quick fixes delete comments and compiler directives
* Methods in parent classes with different parameter names were ignored, triggering false positives

This PR addresses these issues.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,10 +26,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improve AST modeling around property specifiers.
 - Improve semantic analysis around property specifiers.
 - Exclude properties with `implements` specifiers in `UnusedProperty`.
+- Improve `RedundantInheritedCheck`.
 
 ### Fixed
 
 - Overly permissive parsing rules around `string` and `file` types.
+- `RedundantInheritedCheck` removing comments and compiler directives with quick fixes.
```
Comment (Cirras) [inline CHANGELOG.md:34]:
```suggestion

### Fixed

- Overly permissive parsing rules around `string` and `file` types.
- False positives around `message` handler methods in `RedundantInheritedCheck`.
- Quick fixes removing comments and compiler directives in `RedundantInheritedCheck`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]

File: delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java [hunk 2]
```diff
@@ -76,25 +77,32 @@ private static Stream<Type> concreteParentTypesStream(Type type) {
...
+
+    if (parent.getRoutineKind() == RoutineKind.CONSTRUCTOR
+        || parent.getRoutineKind() == RoutineKind.DESTRUCTOR) {
+      // An instance constructor or destructor cannot inherit from a class constructor or
+      // destructor
+      return child.isClassInvocable();
+    }
+
+    // Any other type of invocable can inherit from a class invocable
+    return true;
```
Comment (Cirras) [inline delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java:105]: As far as I can tell, this change isn't relevant to improving `RedundantInheritedCheck`.

File: delphi-checks/src/test/java/au/com/integradev/delphi/checks/RedundantInheritedCheckTest.java [hunk 3]
```diff
@@ -123,6 +123,64 @@ void testOverridingGrandparentShouldNotAddIssue() {
         .verifyNoIssues();
   }
 
+  @Test
+  void testOverridingMethodWithDifferentParamNamesShouldNotAddIssues() {
```
Comment (Cirras) [inline delphi-checks/src/test/java/au/com/integradev/delphi/checks/RedundantInheritedCheckTest.java:127]:
```suggestion
  void testOverridingMethodWithDifferentParameterNamesShouldNotAddIssues() {
```

**Review by Cirras — CHANGES_REQUESTED** [review]

File: delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java (hunk 2 above)
Comment (jgardn3r) [reply delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java:105]: This change alters the way parameters are equated, i.e., parameter lists are now equal if the parameter names are different.

**Comment by Cirras** [comment]
Happy to approve after the merge conflict is resolved.

File: CHANGELOG.md [hunk 4]
```diff
@@ -30,6 +30,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Overly permissive parsing rules around `string` and `file` types.
+- Quick fixes removing comments and compiler directives in `RedundantInheritedCheck`.
+- False positives around `message` handler methods in `RedundantInheritedCheck`.
```
Comment (Cirras) [inline CHANGELOG.md:34]: Oops, missed one thing. The rule key doesn't include the word `Check`.

```suggestion
- Quick fixes removing comments and compiler directives in `RedundantInherited`.
- False positives around `message` handler methods in `RedundantInherited`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]

**Comment by Cirras** [comment]
Also a suggestion for the commit messages:
- Improve `RedundantInherited` quick fixes around comments
- Fix FPs in `RedundantInherited`

File: CHANGELOG.md [hunk 5]
```diff
@@ -31,6 +31,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Overly permissive parsing rules around `string` and `file` types.
 - Quick fixes removing comments and compiler directives in `RedundantInherited`.
+- False positives around `message` handler methods in `RedundantInherited`.
```
Comment (Cirras) [inline CHANGELOG.md:34]:
```suggestion
- False positives around `message` handler methods in `RedundantInherited`.
- False positives when parameter names were mismatched to the inherited method in `RedundantInherited`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]
This time for sure!

**Review by Cirras — APPROVED** [review]
🎉

**Pull request merged**